Move the embedded python interpreter sys.argv fix into the python backend

irasit opened the pull request:
### Summary
Move the `sys.argv` fix for embedded python interpreters (originally added in #375) from `loader.py` to `executor.py` in the python backend.

This is necessary because the C++ PythonBridge uses the python executor directly and bypasses `loader.py`.

More details in #374

### Test Plan
Added a new test that confirms `sys.argv` is available from within python models.

_Note: This description was edited by @VivekPanyam to add details and improve clarity_

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/neuropod?pullRequest=393) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/uber/neuropod?pullRequest=393) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/neuropod?pullRequest=393) it.</sub>

irasit commented:
> Can you remove the code you added to `loader.py` in your previous PR?
> 
> https://github.com/uber/neuropod/blob/4f5764af8f29d1ca241f9576cf4ea5bae5c025dd/source/python/neuropod/loader.py#L23-L27
> 
> We shouldn't include that code in both the loader and the python backend.
> 
> Also, can you update the PR description to match the default format (`Summary` and `Test Plan`)?

Removed the fix from loader, and updated the comment.

irasit commented:
> Also it looks like lint failed (see https://neuropod.ai/developing/#lint-and-static-analysis).
> 
> (You'll need to sign the CLA as well - see the first comment on this PR)

Test case is added. Also remove the version check so it will also fix py2.

irasit commented:
> Looks good! Can you fix lint and sign the CLA (see my comment above: [#393 (review)](https://github.com/uber/neuropod/pull/393#pullrequestreview-436132914))?

Fixed the lint.

VivekPanyam commented:
_Note: Before merging, I edited the PR title, description, and test plan to be more detailed_

cc @irasit
